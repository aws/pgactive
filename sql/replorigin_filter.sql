SELECT * FROM public.bdr_regress_variables()
\gset

\c :writedb1

SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE TABLE public.origin_filter (
   id integer primary key not null,
   n1 integer not null
);
$DDL$);

SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);

-- Simulate a write from some unknown peer node by defining a replication
-- origin and using it in our session. We must forward this write.

INSERT INTO origin_filter(id, n1) VALUES (1, 1);

SELECT pg_replication_origin_create('demo_origin');

SELECT pg_replication_origin_session_is_setup();

INSERT INTO origin_filter(id, n1) VALUES (2, 2);

SELECT pg_replication_origin_session_setup('demo_origin');

SELECT pg_replication_origin_session_is_setup();

INSERT INTO origin_filter(id, n1) VALUES (3, 3);

BEGIN;
SELECT pg_replication_origin_xact_setup('1/1', current_timestamp);
INSERT INTO public.origin_filter(id, n1) values (4, 4);
COMMIT;

SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);

SELECT * FROM origin_filter ORDER BY id;

\c :writedb2

SELECT pg_replication_origin_session_is_setup();

SELECT * FROM origin_filter ORDER BY id;

\c :writedb1

SELECT pg_replication_origin_session_is_setup();

SELECT bdr.bdr_replicate_ddl_command($DDL$
    DROP TABLE public.origin_filter;
$DDL$);

SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);

SELECT pg_replication_origin_drop('demo_origin');
