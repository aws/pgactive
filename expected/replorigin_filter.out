SELECT * FROM public.bdr_regress_variables()
\gset
\c :writedb1
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE TABLE public.origin_filter (
   id integer primary key not null,
   n1 integer not null
);
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

-- Simulate a write from some unknown peer node by defining a replication
-- origin and using it in our session. We must forward this write.
INSERT INTO origin_filter(id, n1) VALUES (1, 1);
SELECT pg_replication_origin_create('demo_origin');
 pg_replication_origin_create 
------------------------------
                            3
(1 row)

SELECT pg_replication_origin_session_is_setup();
 pg_replication_origin_session_is_setup 
----------------------------------------
 f
(1 row)

INSERT INTO origin_filter(id, n1) VALUES (2, 2);
SELECT pg_replication_origin_session_setup('demo_origin');
 pg_replication_origin_session_setup 
-------------------------------------
 
(1 row)

SELECT pg_replication_origin_session_is_setup();
 pg_replication_origin_session_is_setup 
----------------------------------------
 t
(1 row)

INSERT INTO origin_filter(id, n1) VALUES (3, 3);
BEGIN;
SELECT pg_replication_origin_xact_setup('1/1', current_timestamp);
 pg_replication_origin_xact_setup 
----------------------------------
 
(1 row)

INSERT INTO public.origin_filter(id, n1) values (4, 4);
COMMIT;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

SELECT * FROM origin_filter ORDER BY id;
 id | n1 
----+----
  1 |  1
  2 |  2
  3 |  3
  4 |  4
(4 rows)

\c :writedb2
SELECT pg_replication_origin_session_is_setup();
 pg_replication_origin_session_is_setup 
----------------------------------------
 f
(1 row)

SELECT * FROM origin_filter ORDER BY id;
 id | n1 
----+----
  1 |  1
  2 |  2
  3 |  3
  4 |  4
(4 rows)

\c :writedb1
SELECT pg_replication_origin_session_is_setup();
 pg_replication_origin_session_is_setup 
----------------------------------------
 f
(1 row)

SELECT bdr.bdr_replicate_ddl_command($DDL$
    DROP TABLE public.origin_filter;
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

SELECT pg_replication_origin_drop('demo_origin');
 pg_replication_origin_drop 
----------------------------
 
(1 row)

